agency/agent-history-appender.go

```diff
@@ -35,10 +37,13 @@ func (agentState *GeneralAgentInfo) historyAppender() {
 				agentState.History = append(agentState.History, message)
 				atomic.AddInt32(&agentState.historySize, 1)
 			}
-
-			_, _ = agentState.Server.RunRequest(&cmds.ClientRequest{
-				WriteMessagesTrace: []*engines.Message{message},
-			}, 120*time.Second, os_client.REP_IO)
+		case message = <-agentState.systemWriterChannel:
 		}
+		runClientRequest(agentState, message)
 	}
 }
+func runClientRequest(agentState *GeneralAgentInfo, message *engines.Message) {
```

Comment:
The idea of two distinct channels seems great! 

I'm writing about this only because there are two more serious questions in the comments below, and since there's a small chance you might decide to make some corrections while responding to those questions, perhaps you could also address this issue:

I suggest renaming the function to `writeMessageTrace` and calling it from the select twice for improved readability. But really, that's just a minor detail for aesthetics. This will, of course, eliminate the need for the separately declared variable `message`. 

Again, this question isn't critical; it's more about enhancing the code's readability in the future.

Reply:
This is important, thanks. Poor naming on my part.